File: src/test/java/io/vertx/support/CallbackProvider.java
package io.vertx.support;

import io.vertx.codegen.annotations.VertxGen;
import io.vertx.core.AsyncResult;
import io.vertx.core.Future;
import io.vertx.core.Handler;

/**
 * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
 */
@VertxGen
public interface CallbackProvider {

  static void callbackWithSuccess(Handler<AsyncResult<String>> callback) {
    throw new AssertionError("stub");
  }

  public static void callbackWithFailure(Handler<AsyncResult<String>> callback) {
    throw new AssertionError("stub");
  }
}

File: src/test/resources/support-js/callback_provider.js
module.exports = {
    callbackWithSuccess: function(callback) {
        callback("hello", null);
    },
    callbackWithFailure: function(callback) {
        var future = Packages.io.vertx.core.Future.failedFuture("oh no");
        callback(null, future.cause());
    },
    callbackWithString: function(callback) {
        callback("world");
    }
};